code/WorkInProgress/TempEngine.dm

```diff
@@ -444,16 +546,28 @@
 		if(src.variant_b)
 			if(src.variant_b in list("A","B","C","D"))
 				src.generator_flags |= TEG_HIGH_TEMP
+				instructions_footnote += {"<i>Note: Thermo-Electric Generator utilizes experimental alloys optimized for temperatures above 4358K.</i><br>"}
+				variant_description = "An experimental Thermo-Electric Generator! Based on the conductive material it looks like it may be for extremely high temperatures. "
 			else if(src.variant_b in list("E","F","G","H"))
 				src.generator_flags |= TEG_LOW_TEMP
+				instructions_footnote += {"<i>Note: Thermo-Electric Generator utilizes experimental alloys optimized for temperatures below 1550K.</i><br>"}
+				variant_description = "An experimental Thermo-Electric Generator! Based on the conductive material it looks like it may be designed for cooler temperatures. "
 			else
 				// Reassign variant_b to null so unsupported variants aren't shown to players
 				// to avoid confusion
 				src.variant_b = null
+				variant_description = null
 
 		src.circ1?.assign_variant(prepend_serial_num, src.variant_a, src.variant_b)
 		src.circ2?.assign_variant(prepend_serial_num, src.variant_a, src.variant_b)
 
+
+		SPAWN_DBG(0) // Offload this action...
+			for(var/obj/item/paper/engine/instructions in range(40,src))
```

Comment:
40? no.

Reply:
better handled with a by_type imo

Reply:
Some CE offices are oddly distant from TEG, I'll try out by_type(). Honestly did not like my solution besides it being able to be offloaded by spawn at startup.

Reply:
Added by_type tracking to TEG instructions.
Limited updating to current zlevel to account for prefab (that would have been an awkward bug to find)

Speeding up the search feels meh for what should occur once but round but it decouples the implementation from map design which is WAY better. Good call.